torch/_prims/utils.py

```diff
@@ -603,11 +609,11 @@ def is_grad_dtype(dtype: torch.dtype) -> bool:
 
 
 def corresponding_real_dtype(dtype: torch.dtype) -> torch.dtype:
-    return _complex_to_real_dtype_map[dtype]
+    return _complex_to_real_dtype_map.get(dtype, dtype)
 
 
 def corresponding_complex_dtype(dtype: torch.dtype) -> torch.dtype:
-    return _real_to_complex_dtype_map[dtype]
+    return _real_to_complex_dtype_map.get(dtype, dtype)
```

Comment:
Returning torch.long for the "corresponding complex dtype" to long seems wrong, though? Where is this needed? The semantics of "float types should be complex but integer types should stay integer types" seems odd

Reply:
We should probably just assert it's in the map if we want to improve the error message over the look-up error one would currently get

Reply:
fwiw, and at risk at stating the obvious, the point of this get is for this function to be the identity on fp types.

Reply:
(or we could add complex->complex mappings, too, and just check keys)

Reply:
Identity on COMPLEX dtypes, you mean?

Reply:
yes, sorry, the identity on complex dtypes here and the identity  on real dtypes in the function before this one.

Reply:
Yeah, just reverting this change and the few lines that depend on it should be fine